src/hooks/common/useInfiniteScroll.ts

```
@@ -0,0 +1,56 @@
+ * <div ref={sentinelRef} />
+ */
+export function useInfiniteScroll<T extends HTMLElement = HTMLElement>({
+  hasMore,
+  isLoading,
+  onLoadMore,
+  rootMargin = "200px",
+}: UseInfiniteScrollOptions) {
+  const ref = useRef<T>(null);
+  // onLoadMore가 매 렌더 새로 만들어져도 effect가 재등록되지 않도록 ref로 고정
+  const onLoadMoreRef = useRef(onLoadMore);
+  onLoadMoreRef.current = onLoadMore;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/hooks/common/useInfiniteScroll.ts:38:3
  36 |   // onLoadMore가 매 렌더 새로 만들어져도 effect가 재등록되지 않도록 ref로 고정
  37 |   const onLoadMoreRef = useRef(onLoadMore);
> 38 |   onLoadMoreRef.current = onLoadMore;
     |   ^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  39 |
  40 |   useEffect(() => {
  41 |     const el = ref.current;

(react-hooks/refs)